Documentation/api.md

```diff
@@ -1,5 +1,27 @@
 # etcd API
 
+- [Key Space Operations](#key-space-operations)
```

Comment:
i still do not quite like the big tables actually. but it might be only me.

/cc @yichengq @barakmich